core/src/main/java/org/elasticsearch/index/query/IndicesQueryBuilder.java

```diff
@@ -31,48 +39,67 @@
 
     public static final String NAME = "indices";
 
-    private final QueryBuilder queryBuilder;
+    /** The default query to be executed on non matching indices */
+    public static final QueryBuilder DEFAULT_NO_MATCH_QUERY = QueryBuilders.matchAllQuery();
+
+    private final QueryBuilder innerQuery;
 
     private final String[] indices;
 
-    private String sNoMatchQuery;
-    private QueryBuilder noMatchQuery;
+    private QueryBuilder noMatchQuery = DEFAULT_NO_MATCH_QUERY;
 
     static final IndicesQueryBuilder PROTOTYPE = new IndicesQueryBuilder(null);
 
-    public IndicesQueryBuilder(QueryBuilder queryBuilder, String... indices) {
-        this.queryBuilder = queryBuilder;
+    public IndicesQueryBuilder(QueryBuilder innerQuery, String... indices) {
+        this.innerQuery = innerQuery;
```

Comment:
Not sure, but maybe we can add null-check for innerQuery here like in other nested query builders (e.g. BoostingQueryBuilder)? That would avoid potential NPEs or null checks in doXContent later.